src/LogicGate1.java

```diff
@@ -0,0 +1,190 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Map;
+
+/**
+ * {@code LogicGate} represented as a connectable node. Interacts with other gates.
+ * 
+ * 
+ * 
+ * @convention this is empty and not connected to any other gates or this connects to and/or is connected to by other gates and has a full truth table 
+ * @correspondance this is represented as a nonhierarchical connection of gates
+ */
+public class LogicGate1 {
+
+    /*
+     * The logic gates leading into this one.
+     */
+    private ArrayList<LogicGate1> inputs = new ArrayList<>();
+
+    /**
+     * All of the gates in the chain of this.
+     */
+    private HashSet<LogicGate1> allGates = new HashSet<>();
+
+    /**
+     * The truth table {@code this} consults.
+     */
+    Map<String, String> truthTable;
+
+    /**
+     * The name of {@code this}. 
+     */
+    String name;
+
+    /**
+     * The value of {@code this} ONLY if this is a true or false gate. This does not store the temporary value of a gate with a truth table.
+     */
+    String value;
+
+    private void createNewRep(){
+        this.name = "Default";
+        this.inputs = null;
+        this.truthTable = null;
+        this.allGates = null;
+    }
+
+    /**
+     * Constructor from a truth table.
+     *
+     * @param inputTruthTable
+     * @param gateNameString
+     */
+    public LogicGate(Map<String, String> inputTruthTable,
+            String gateNameString) {
+        this.createNewRep();
+        this.truthTable = inputTruthTable;
+        this.name = gateNameString;
+    }
+
+    /**
+     * Constructor for a TRUE or FALSE gate.
+     *
+     * @param value
+     * @param gateNameString
+     */
+    public LogicGate(String value, String gateNameString) {
+        this.createNewRep();
+        this.value = value;
+        this.name = gateNameString;
+    }
+
+    @Override
+    public final void clear() {
```

Comment:
You don't have a no-argument constructor, so this might put your object in a weird state.